doc/dev/TESTING.md

```diff
@@ -1,77 +1,117 @@
 
 ## Running the tests
 
-To run the tests, just use ctest from the build directory:
+To run all tests, build the `f3d` binary with `make` and then run `ctest` from the build directory:
 
 ```
+make
 ctest
 ```
 
-To run a specific test, use the -R ctest option:
+To run a specific test, use the `ctest -R <testname>` option:
 
 ```
 ctest -R PLY
 ```
 
 ## Testing architecture
 
-There is multiple layers of tests to ensure F3D test coverage is as high as possible
- - Application layer
+There are multiple layers of tests to ensure that testing covers all aspects of the application. The layers of the application are
+ - UI (Application) layer
  - Library layer
  - Bindings layer
  - VTK Extension layer
 
-When contributing to F3D, it is necessary that any new code is covered by at least one layer
-of test, but it could make sense to cover it with more if necessary.
+When contributing to F3D, it is necessary that new code is covered by a test in at least one layer above. Additional tests can be created in other layers, depending on the new functionality / bugfix.
 
 ### Application layer
 
-All application test are just a command line run by ctest using the `f3d` executable.
-ctest then check the output for any failure. Most of these tests are just using the `--output`
-and `--ref` F3D option in order to check if a rendering behave correctly with specific options.
+All application tests are initiated via the `ctest` testing framework.
+`ctest` then runs `f3d` with the `--output` and `--ref` arguments which renders an output image and compares it to a reference image. `f3d` also compares the differences between the 2 images and generates a diff file if there is a difference.
 
-Everything is handled in `application/testing/CMakeLists.txt`.
+All aspects of the application test framework are handled in `application/testing/CMakeLists.txt`.
 
-Usually, adding a test is a simple as adding a line like this one:
+Usually, adding a test is as simple as adding a line like this one:
 
 ```
 f3d_test(NAME TestName DATA datafile.ext ARGS --args-to-test DEFAULT_LIGHTS)
 ```
 
+ where
  - `NAME` should be the name of the test, which must be unique
- - `DATA` should be a file in `testing/data` directory, though adding new file is possible
+ - `DATA` should be a file in `testing/data` directory, though adding a new file is possible
  - `ARGS` should be the F3D options to pass to the f3d executable, if any
  - `DEFAULT_LIGHTS` is expected when performing baselines comparison
 
-Once the new test has been added, configure and build F3D, then run the test once:
+Once the new test has been added, configure and build F3D, then run the test (-VV for verbose output):
 
 ```
-ctest -R TestName
+ctest -R TestName -VV
 ```
 
-The test will fail but an image output will be generated in you build in `Testing/Temporary/TestName.png`.
+The test will fail but an image output will be generated in the build directory, namely `build/Testing/Temporary/TestName.png`.
```

Comment:
you cannot assume the build directory is named "build"

Reply:
I assumed it would be `build` as that is what is specified in the getting_started doc.
Point noted, I think the developer will know what he has called his "build" directory.